src/Component/Filter/FilterTree/FilterTree.tsx

```diff
@@ -0,0 +1,567 @@
+    filter: ['==', '', null]
+  };
+
+  constructor(props: FilterTreeProps) {
+    super(props);
+    this.state = {
+      expandedKeys: []
+    };
+  }
+
+  /**
+   * Changehandler for ComparsionFilters.
+   *
+   */
+  onComparisonFilterChange = (filter: GsFilter, position: string) => {
+    const {
+      filter: rootFilter
+    } = this.props;
+    if (filter === rootFilter) {
+      this.props.onFilterChange(filter);
+    }
+
+    let newFilter = [...rootFilter];
+    if (position === '') {
+      newFilter = filter;
+    } else {
+      _set(newFilter, position, filter);
+    }
+
+    this.props.onFilterChange(newFilter);
+  }
+
+  /**
+   * Creates a TreeNode for a given filter at the given position.
+   *
+   * @return Tree.TreeNode
+   */
+  getNodeByFilter = (filter: GsFilter, position: string = ''): any => {
+    const {
+      internalDataDef
+    } = this.props;
+    const operator = filter[0];
+
+    let showRemoveButton = false;
+
+    const parentPosition = position.substring(0, position.length - 3);
+    const parentFilter = position === '' ? null : this.getFilterAtPosition(parentPosition);
+
+    if (parentFilter) {
+      const parentOperator = parentFilter[0];
+      if (parentOperator === '&&' || parentOperator === '||') {
+        showRemoveButton = true;
+      }
+    }
+
+    const addFilterMenu = (
+      <Menu onClick={({key}) => this.addFilter(position, key)}>
+        <Menu.Item key="and">And-Filter</Menu.Item>
+        <Menu.Item key="or">Or-Filter</Menu.Item>
+        <Menu.Item key="not">Not-Filter</Menu.Item>
+        <Menu.Item key="comparison">Comparison-Filter</Menu.Item>
+      </Menu>
+    );
+
+    const changeFilterMenu = (
+      <Menu onClick={({key}) => this.changeFilter(position, key)}>
+        <Menu.Item key="and">And-Filter</Menu.Item>
+        <Menu.Item key="or">Or-Filter</Menu.Item>
+        <Menu.Item key="not">Not-Filter</Menu.Item>
+        <Menu.Item key="comparison">Comparison-Filter</Menu.Item>
+      </Menu>
+    );
+
+    const addButton = (
+      <Tooltip title="Add Filter" placement="top">
+        <Dropdown
+          trigger={['click']}
+          overlay={addFilterMenu}
+        >
+          <Button size="small">
+            <Icon type="plus"/>
+          </Button>
+        </Dropdown>
+      </Tooltip>
+    );
+
+    const removeButton = (
+      <Tooltip title="Remove Filter" placement="right">
+        <Button
+          size="small"
+          onClick={() => this.removeFilter(position)}
+        >
+          <Icon
+            type="minus"
+          />
+        </Button>
+      </Tooltip>
+    );
+
+    const changeButton = (
+      <Tooltip title="Change Filter" placement="left">
+        <Dropdown
+          trigger={['click']}
+          overlay={changeFilterMenu}
+        >
+          <Button size="small">
+            <Icon type="filter"/>
+          </Button>
+        </Dropdown>
+      </Tooltip>
+    );
+
+    if (operator === '&&') {
```

Comment:
Minor note: A switch statement could improve the readability and maintainability, here.